tc/core/polyhedral/cuda/mapped_scop.cc

```diff
@@ -230,12 +229,17 @@ bool MappedScop::detectReductions(detail::ScheduleTree* tree) {
   if (!isReductionMember(member, updates, scop())) {
     return false;
   }
-  // Order the init statements (if any) before the update statements
+  // Order the other statements (if any) before the update statements
```

Comment:
Can't we have other statements that are not updates and should be ordered _after_ reduction updates?  Something like
```
O(i)  = 0 for i in 0:N     #order before
O(i) += A(i,j)
O(i)  = O(i) / M             #order after
```

with a schedule
```
band([{S1[i]->[i]; S2[i,j]->[i]; S3[i]->[i]}; {S1[i]->[0]; S2[i,j]->[j]; S3[i]->[M]}])
  sequence()
    filter(S1[i])
    filter(S2[i,j])
    filter(S3[i])
```